File: 240228/총점 비교/compare-total-points.js
class Student {
    constructor(name, a, b, c) {
        this.name = name;
        this.a = a;
        this.b = b;
        this.c = c;
    }
}

const fs = require("fs");
let [n, ...data] = fs.readFileSync(0).toString().trim().split("\n");

const students = []
for(let d of data) {
    const s = d.trim().split(' ')
    students.push(new Student(s[0],Number(s[1]),Number(s[2]),Number(s[3])))
}

students.sort((a,b) => (a.a + a.b + a.c) - (b.a + b.b + b.c))
students.forEach(s => {
    console.log(`${s.name} ${s.a} ${s.b} ${s.c}`)
})

File: 240228/총점 비교/compare-total-points.py
class Student:
    def __init__(self, name, a, b, c):
        self.name = name
        self.a = a
        self.b = b
        self.c = c

n = int(input())
students = []

for _ in range(n):
    name, a, b, c = tuple(input().split())
    students.append(Student(name, int(a), int(b), int(c)))

students.sort(key = lambda x: x.a + x.b + x.c)

for s in students:
    print(s.name, s.a, s.b, s.c)